src/services/global-descriptor-service.ts

```diff
@@ -0,0 +1,299 @@
+			gas
+		};
+		const value = ((await this.contract.methods
+			.getLatestUnitValue(unit)
+			.call(txOptions)) as unknown) as string;
+
+		return parseInt(value, 10) / this.DECIMAL_OFFSET;
+	}
+
+	/**
+	 * Abstraction for GlobalDescriptors (smart contract) method, getAllAvailableUnitForGlobal(): string[]
+	 * See ./contracts/GlobalDescriptors.sol for the actual contract method
+	 * @param accountId ID of the account sending the request (Local blockchain autogenerates 10 accounts to use)
+	 * @param gas Optional paramter, defaults to 5,000,000. Need gas to perform any sort of operation.
+	 */
+	public async getAllAvailableUnitsForGlobal(
+		accountId: string,
+		gas = 5_000_000
+	) {
+		const txOptions: Tx = {
+			from: accountId,
+			gas
+		};
+		return this.contract.methods.getAllAvailableUnits().call(txOptions);
+	}
+
+	/**
+	 * Abstraction for GlobalDescriptors (smart contract) method, getAllValuesRecordedForUnit(unit: string): number[]
+	 * See ./contracts/GlobalDescriptors.sol for the actual contract method
+	 * @param accountId ID of the account sending the request (Local blockchain autogenerates 10 accounts to use)
+	 * @param unit A unit such as lb, cm, miles, kilometer, etc
+	 * @param gas Optional paramter, defaults to 5,000,000. Need gas to perform any sort of operation.
+	 */
+	public async getAllValuesRecordedForUnit(
+		accountId: string,
+		unit: string,
+		gas = 5_000_000
+	): Promise<IDescriptor[]> {
+		const txOptions: Tx = {
+			from: accountId,
+			gas
+		};
+		const stringArray = await this.contract.methods
+			.getAllUnitValues(unit)
+			.call(txOptions);
+		return stringArray.map(val => {
+			return {
+				unit,
+				value:
+					GlobalDescriptorService.BNToNumber(val.unitValue) /
+					this.DECIMAL_OFFSET,
+				longitude:
+					GlobalDescriptorService.BNToNumber(val.longitude) /
+					this.DECIMAL_OFFSET,
+				latitude:
+					GlobalDescriptorService.BNToNumber(val.latitude) /
+					this.DECIMAL_OFFSET,
+				unixTimestamp: GlobalDescriptorService.BNToNumber(val.time)
+			};
+		});
+	}
+
+	public async getPaginatedValuesRecordedForUnit(
+		accountId: string,
+		unit: string,
+		start = 0,
+		count = 50,
+		gas = 5_000_000
+	): Promise<IDescriptor[]> {
+		const stringArray = await this.contract.methods
+			.getPaginatedUnitValues(unit, start, count)
+			.call({ from: accountId, gas });
+		return stringArray.map(val => {
+			return {
+				unit,
+				value:
+					GlobalDescriptorService.BNToNumber(val.unitValue) /
+					this.DECIMAL_OFFSET,
+				longitude:
+				GlobalDescriptorService.BNToNumber(val.longitude) /
+					this.DECIMAL_OFFSET,
+				latitude:
+				GlobalDescriptorService.BNToNumber(val.latitude) /
+					this.DECIMAL_OFFSET,
+				unixTimestamp: GlobalDescriptorService.BNToNumber(val.time)
+			};
+		});
+	}
+
+
+
+
+	/**
+	 * Gets all the accounts in the blockchain and returns the ID of the account at
+	 * a certain index in the returned array.
+	 */
+	public async getAccountAtIndex(index: number): Promise<string> {

```

Comment:
Lets get rid of this, we already have one and it doesn't provide anything for this class itself. If you want, you can move something like this into src/utilities from both global and user descriptors.